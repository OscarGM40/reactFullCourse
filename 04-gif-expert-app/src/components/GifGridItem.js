import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({title,url}) => {
   
   return (
      <div className="card animate__animated animate__bounce">
         <img src={url} alt={title}></img>
         <p>{title}</p>
      </div>
   )
}

GifGridItem.propTypes = {

   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired,
}

export default GifGridItem;


//Tarea 
// 1- Configurar Enzyme
// 2- Configurar Enzyme to JSON
// 3- debe de mostrar el componente correctamente
// * shallow + wrapper.toMatchSnapShot()
