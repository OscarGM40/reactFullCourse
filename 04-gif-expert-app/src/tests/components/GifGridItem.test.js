import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';


describe('Pruebas en GifGridItem', () => {
   
   const props = { 
      title:"Un titulo",
      url:"http://www.w3.org/1999/"
   }
   const wrapper = shallow(<GifGridItem { ...props }/>);
   
   test('debe de mostrar el componente', () => {
      
      expect(wrapper).toMatchSnapshot();

   })

   test('debe de tener un parrafo con el titulo', () => {
      
      const parrafo = wrapper.find('p').text().trim();
      const { title } = props;
      expect(parrafo).toBe(title) 

   })
   
   test('debe de tener imagen igual al url y al de los props', () => {
      
      const img = wrapper.find('img');
      //console.log(img.html(),'metodo html')
      //console.log(img.props().src,'metodo props.atributo')
      //console.log(img.prop('alt'),'metodo prop(atributo)')

      expect(img.prop('src')).toBe(props.url);
      expect(img.prop('alt')).toBe(props.title);

   })

   test('debe de tener cierta clase CSS', () => {
      
      const div = wrapper.find('div');
      //console.log(div.props().className,'props del div');
      const clases = div.props().className.split(' ');

      expect(clases[2]).toBe("animate__bounce");
      expect(div.prop('className').includes('animate__bounce')).not.toBe(false);
      expect(div.hasClass('animate__bounce')).toBeTruthy();
   })
   
   
   
})
