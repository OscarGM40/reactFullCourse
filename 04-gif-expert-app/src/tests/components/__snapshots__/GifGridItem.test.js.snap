// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe de mostrar el componente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo"
    src="http://www.w3.org/1999/"
  />
  <p>
    Un titulo
  </p>
</div>
`;
